Python/pylifecycle.c

```diff
@@ -1257,6 +1257,20 @@ Py_FinalizeEx(void)
     PyThreadState *tstate = _PyRuntimeState_GetThreadState(runtime);
     PyInterpreterState *interp = tstate->interp;
 
+    // Finalize sub-interpreters.
+    runtime->interpreters.finalizing = 1;
```

Comment:
We need to make sure this is set back to 0 at some point, else the runtime cannot be re-initialized.  The name "finalizing" implies that it would be set to 0 once finalization is complete.  However, it would at least have to be `_PyRuntime_Initialize()`.

FWIW, using a `ready`/`available` flag as I've suggested above would make the semantics more obvious.

Reply:
Have updated based on your suggestions above.